### ISSUE TITLE
[A11Y] Category navigator accessibility

### ISSUE THREAD

**Issue by foxbunny** [indico/indico#5917]
**Describe the bug**

The category navigator has multiple accessibility issues:

- Not announced as a dialog to the screen reader
- Keyboard navigation is broken
- Navigation (action) button is not visible by default

**To Reproduce**

Scenario A:

1. Go to main page
2. Activate "Navigate"
3. Observe that screen reader does not announce the presence of a dialog

Scenario B:

1. Go to main page
2. Activate "Navigate"
3. Try tabbing into one of the categories
4. Observe that navigation skips over the category list and goes to "Close" button

Scenario C:

1. Go to main page
2. Activate "Navigate"
3. Observe that "Navigate to >" is not shown for any categories

**Expected behavior**

Scenario A: the dialog is correctly announced as a dialog
Scenario B: it should be possible to use the tab navigation to access categories
Scenario C: the "Navigate to >" (or an alternative cue) should be present without hover

**Additional context**

- https://talk.getindico.io/t/keyboard-improve-keyboard-interaction-of-buttons-in-the-category-navigation/3209
- https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html
- https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html
